pkg/reconciler/kubernetes/tektonhub/tektonhub.go

```diff
@@ -0,0 +1,595 @@
+			_, err = r.kubeClientSet.CoreV1().ConfigMaps(namespace).Create(ctx, configMap, metav1.CreateOptions{})
+			if err != nil {
+				logger.Error(err)
+				th.Status.MarkApiDependencyMissing(fmt.Sprintf("%s configMap is missing", apiConfigMapName))
+				return err
+			}
+			return nil
+		}
+		if err == errKeyMissing {
+			th.Status.MarkApiDependencyMissing(fmt.Sprintf("%s configMap is missing the keys", apiConfigMapName))
+			return err
+		} else {
+			logger.Error(err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *Reconciler) getSecretForHub(ctx context.Context, name, targetNs string, keys []string) (*corev1.Secret, error) {
+	secret, err := r.kubeClientSet.CoreV1().Secrets(targetNs).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	allKeys := true
+	for _, key := range keys {
+		if _, ok := secret.Data[key]; !ok {
+			allKeys = false
+			break
+		}
+	}
+
+	if !allKeys {
+		return nil, errKeyMissing
+	}
+
+	return secret, nil
+}
+
+func (r *Reconciler) getConfigMapForHub(ctx context.Context, name, targetNs string, keys []string) (*corev1.ConfigMap, error) {
+	configMap, err := r.kubeClientSet.CoreV1().ConfigMaps(targetNs).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	allKeys := true
+	for _, key := range keys {
+		if _, ok := configMap.Data[key]; !ok {
+			allKeys = false
+			break
+		}
+	}
+
+	if !allKeys {
+		return nil, errKeyMissing
+	}
+
+	return configMap, nil
+}
+
+func createApiConfigMap(name, namespace string, th *v1alpha1.TektonHub) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app": "api",
+			},
+		},
+		Data: map[string]string{
+			"CONFIG_FILE_URL": th.Spec.Api.HubConfigUrl,
+		},
+	}
+}
+
+func createDbSecret(name, namespace string, existingSecret *corev1.Secret) *corev1.Secret {
+	s := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app": "db",
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+
+	if existingSecret != nil && existingSecret.Data != nil {
+		s.Data = existingSecret.Data
+	}
+
+	s.StringData = make(map[string]string)
+
+	if s.Data["POSTGRES_DB"] == nil || len(s.Data["POSTGRES_DB"]) == 0 {
+		s.StringData["POSTGRES_DB"] = "hub"
+	}
+
+	if s.Data["POSTGRES_USER"] == nil || len(s.Data["POSTGRES_USER"]) == 0 {
+		s.StringData["POSTGRES_USER"] = "hub"
+	}
+
+	if s.Data["POSTGRES_PASSWORD"] == nil || len(s.Data["POSTGRES_PASSWORD"]) == 0 {
+		s.StringData["POSTGRES_PASSWORD"] = "hub"
+	}
+
+	if s.Data["POSTGRES_PORT"] == nil || len(s.Data["POSTGRES_PORT"]) == 0 {
+		s.StringData["POSTGRES_PORT"] = "5432"
+	}
+
+	return s
+}
+
+func (r *Reconciler) applyManifest(ctx context.Context, manifestLocation string, th *v1alpha1.TektonHub, installerSetName, version, prefixName string) error {
```

Comment:
misleading name `applyManifest`. 
i was expecting to see a `m.Apply()`. I understand that createInstallerSet might be already in use. May be make this name `setupAndCreateInstallerSet` or some other name that reveals the intent(behavior) of this method. 

Explicit and boring is good. 🧑‍💻